File: gamehub-server/src/main/java/com/gamehub/service/auth/AuthService.java
package com.gamehub.service.auth;

import com.gamehub.api.model.auth.AuthRequestModel;
import com.gamehub.api.model.auth.AuthResponseModel;

public interface AuthService {
    AuthResponseModel doLogin(final AuthRequestModel authRequestModel);
    void doLogout(final String jwtToken);
}

File: gamehub-android/app/src/main/java/com/gamehub/service/AuthService.kt
package com.gamehub.service

import com.gamehub.model.auth.AuthRequestModel
import com.gamehub.model.auth.AuthResponseModel
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.POST

interface AuthService {
    @POST("/api/auth/login")
    fun login(@Body request: AuthRequestModel): Call<AuthResponseModel>
}